pkg/virtctl/ssh/knownhosts_test.go

```diff
@@ -0,0 +1,58 @@
+package ssh
+
+import (
+	"bufio"
+	"crypto/rand"
+	"crypto/rsa"
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/crypto/ssh"
+)
+
+var _ = Describe("Known Hosts", func() {
+
+	var knowHostFile string
+	BeforeEach(func() {
+		tmpDir := GinkgoT().TempDir()
+		knowHostFile = filepath.Join(tmpDir, "knownhosts")
+		f, err := os.Create(knowHostFile)
+		Expect(err).ToNot(HaveOccurred())
+		_ = f.Close()
+	})
+
+	It("should be added with a newline", func() {
+		publicKey := newPublicKey()
+		Expect(addHostKey(knowHostFile, "host1", publicKey)).To(Succeed())
+		Expect(addHostKey(knowHostFile, "host2", publicKey)).To(Succeed())
+		Expect(addHostKey(knowHostFile, "host3", publicKey)).To(Succeed())
+
+		Expect(numberOfLines(knowHostFile)).To(Equal(3))
+	})
+})
+
+func newPublicKey() ssh.PublicKey {
+	privateKey, err := rsa.GenerateKey(rand.Reader, 1024)
+	Expect(err).ToNot(HaveOccurred())
+	// generate and write public key
+	pub, err := ssh.NewPublicKey(&privateKey.PublicKey)
+	Expect(err).ToNot(HaveOccurred())
+	return pub
+}
+
+func numberOfLines(knowHostFile string) int {
+	f, err := os.Open(knowHostFile)
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
I am not fond of asserting errors within helper functions - hence, I'd prefer if you returned the error.

Having said that, I'm fully aware that's opinionated; if you want to preserve this signature, I would prefer if you used `ExpectWithOffset`, since it would allow Gomega's error message to refer to the calling line in the test (as opposed to the line in the helper function): https://github.com/onsi/gomega/blob/d01bc22577e88a1c7afa2448e81ee8e43a598d28/gomega_dsl.go#L210

Does that make sense to you @rmohr ?

Reply:
Proper helpers are notoriously always a little bit verbose and I can't directly see what they are doing. Therefore I prefer not to add too many of them when dealing with a single test, but yes, makes sense. Done.